src/libraries/StringParser.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {VmSafe} from "forge-std/Vm.sol";
+import {LibString} from "@solady/utils/LibString.sol";
+
+/// @title StringParser
+/// @notice Library for parsing string representations of decimal numbers
+library StringParser {
+    using LibString for string;
+
+    /// @notice Struct to hold parsed amount components
+    struct AmountComponents {
+        uint256 wholePart;
+        uint256 decimalPart;
+        uint8 decimalPlaces;
+        bool hasDecimals;
+    }
+
+    /// @notice Validates that the amount string has a valid format
+    /// @param components The split components of the amount string
+    /// reverts if components are invalid
+    function validateAmountFormat(string[] memory components) internal pure {
+        // Check for invalid format (more than one decimal point)
+        require(components.length <= 2, "invalid amount");
+
+        // If there's a decimal part, ensure it's not empty
+        if (components.length == 2) {
+            require(bytes(components[1]).length != 0, "decimals cannot be 0");
+        }
+    }
+
+    /// @notice Checks if the amount represents a non-zero value
+    /// @param wholePart The whole number part of the amount
+    /// @param decimalPart The decimal part of the amount
+    /// @return True if either part results in a non-zero value
+    function isNonZeroAmount(uint256 wholePart, uint256 decimalPart) internal pure returns (bool) {
+        return wholePart > 0 || decimalPart > 0;
+    }
+
+    /// @notice Parses a string amount into its components
+    /// @param amount The amount string to parse
+    /// @return components The parsed amount components
+    function parseAmountComponents(string memory amount) internal pure returns (AmountComponents memory components) {
+        VmSafe vm = VmSafe(address(uint160(uint256(keccak256("hevm cheat code")))));
```

Comment:
Can you move this declaration to the top level of this file? Like this: https://github.com/ethereum-optimism/superchain-ops/blob/main/src/libraries/AccountAccessParser.sol#L46-L47 

It's what we're doing in other files and it's easy for me to know that this file requires vm access.